src/commit.c

```diff
@@ -98,6 +98,46 @@ Commit_gpg_signature__get__(Commit *self)
 }
 
 
+PyDoc_STRVAR(Commit_message_trailers__doc__,
+    "Returns commit message trailers (e.g., Bug: 1234) as a dictionary."
+);
+
+PyObject *
+Commit_message_trailers__get__(Commit *self)
+{
+    git_message_trailer_array gmt_arr;
+    int i, trailer_count, err;
+    PyObject *dict;
+    PyObject *py_val;
+    const char *message = git_commit_message(self->commit);
+    const char *encoding = git_commit_message_encoding(self->commit);
+
+    err = git_message_trailers(&gmt_arr, message);
+    if (err < 0)
+        return Error_set(err);
+
+    dict = PyDict_New();
+    if (dict == NULL)
+        return NULL;
```

Comment:
In this case `git_message_trailer_array_free` is not called.

Reply:
You can just `goto error;`

Reply:
Good catch, thank you!